src/KubernetesClient/Kubernetes.ConfigInit.cs

```diff
@@ -115,12 +115,7 @@ private void InitializeFromConfig(KubernetesClientConfiguration config)
         private void CreateHttpClient(DelegatingHandler[] handlers, KubernetesClientConfiguration config)
         {
 #if NET5_0_OR_GREATER
-            FirstMessageHandler = HttpClientHandler = new SocketsHttpHandler
-            {
-                KeepAlivePingPolicy = HttpKeepAlivePingPolicy.WithActiveRequests,
-                KeepAlivePingDelay = TimeSpan.FromMinutes(3),
-                KeepAlivePingTimeout = TimeSpan.FromSeconds(30),
-            };
+            FirstMessageHandler = HttpClientHandler = config.SocketsHttpHandlerFactory();
 
             HttpClientHandler.SslOptions.ClientCertificates = new X509Certificate2Collection();
 #else
```

Comment:
could you please also add similar factory to `#else` (netstandard2)

Reply:
Ok, but I'll switch to a callback instead, I don't like seeing too much code on the config side